openwisp_controller/config/static/config/js/default_templates.js

```diff
@@ -46,23 +47,81 @@ django.jQuery(function ($) {
                 });
             });
         },
-        bindDefaultTemplateLoading = function (urls) {
+        getRelevantTemplates = function (urls) {
+            var orgID = $('#id_organization').val(),
+                backend = $('#id_config-0-backend').val();
+            if (!orgID || !backend) {
+                return;
+            }
+            // proceed only if an organization and a backend have been selected
+            if (orgID.length === 0 || backend.length === 0) {
+                return;
+            }
+
+            var url = urls[orgID];
+            // get relevant templates of selected org and backend
+            url = url + '?backend=' + backend;
+            $.get(url).done(function (data) {
+                var relevantTemplates = {};
+                $.each(data.templates, function (i, uuid) {
+                    relevantTemplates[uuid] = uuid;
+                });
+                var disabledTemplates =
+                    localStorage.getItem('config-disabled-templates', null);
+                disabledTemplates = JSON.parse(disabledTemplates) || {};
+                $('input.sortedm2m').each(function () {
+                    if (($(this).val() in relevantTemplates) === false) {
```

Comment:
Why not use do `relevantTemplates[$(this).val()]` and put everything in try catch instead? The "in" operator will take more time if number of entries is more in relevantTemplates.